docs/Learning-Environment-Design-Agents.md

```diff
@@ -88,81 +100,135 @@ when certain game or simulation events occur, such as in a turn-based game,
+booleans directly to the observation vector, as well as some common Unity data
+types such as `Vector2`, `Vector3`, and `Quaternion`.
 
 For examples of various state observation functions, you can look at the
 [example environments](Learning-Environment-Examples.md) included in the
 ML-Agents SDK. For instance, the 3DBall example uses the rotation of the
 platform, the relative position of the ball, and the velocity of the ball as its
-state observation. As an experiment, you can remove the velocity components from
-the observation and retrain the 3DBall agent. While it will learn to balance the
-ball reasonably well, the performance of the agent without using velocity is
-noticeably worse.
+state observation.
 
 ```csharp
 public GameObject ball;
 
-private List<float> state = new List<float>();
 public override void CollectObservations(VectorSensor sensor)
 {
+    // Orientation of the cube (2 floats)
     sensor.AddObservation(gameObject.transform.rotation.z);
     sensor.AddObservation(gameObject.transform.rotation.x);
-    sensor.AddObservation((ball.transform.position.x - gameObject.transform.position.x));
-    sensor.AddObservation((ball.transform.position.y - gameObject.transform.position.y));
-    sensor.AddObservation((ball.transform.position.z - gameObject.transform.position.z));
-    sensor.AddObservation(ball.transform.GetComponent<Rigidbody>().velocity.x);
-    sensor.AddObservation(ball.transform.GetComponent<Rigidbody>().velocity.y);
-    sensor.AddObservation(ball.transform.GetComponent<Rigidbody>().velocity.z);
+    // Relative position of the ball to the cube (3 floats)
+    sensor.AddObservation(ball.transform.position - gameObject.transform.position);
+    // Velocity of the ball (3 floats)
+    sensor.AddObservation(m_BallRb.velocity);
+    // 8 floats total
 }
 ```
 
-The feature vector must always contain the same number of elements and
-observations must always be in the same position within the list. If the number
-of observed entities in an environment can vary you can pad the feature vector
-with zeros for any missing entities in a specific observation or you can limit
+As an experiment, you can remove the velocity components from
+the observation and retrain the 3DBall agent. While it will learn to balance the
+ball reasonably well, the performance of the agent without using velocity is
+noticeably worse.
+
+The observations passed to `VectorSensor.AddObservation()` must always contain
+the same number of elements must always be in the same order. If the number
+of observed entities in an environment can vary, you can pad the calls
+with zeros for any missing entities in a specific observation, or you can limit
 an agent's observations to a fixed subset. For example, instead of observing
-every enemy agent in an environment, you could only observe the closest five.
+every enemy in an environment, you could only observe the closest five.
```

Comment:
De-emphasized the "agent" part of "enemy agent" so as not to conflate with multiagent training.